kartothek/io_components/metapartition.py

```diff
@@ -1629,7 +1636,9 @@ def parse_input_to_metapartition(
                 )
             )
     elif isinstance(obj, dict):
-        if isinstance(obj["data"], list):
+        if not obj.get("data"):
+            data = obj
+        elif isinstance(obj["data"], list):
```

Comment:
@fjetter since `indices` and `label` are deprecated, shouldn't we also deprecate this format (`{"data": {"table": df}}`) in favor of `{"table":df}`?

Reply:
what I was going for was raising a deprecation warning

Reply:
Yes, but I'm not sure what the future behaviour will be. Without this a deprecation warning seems premature to me. I don't want to write a deprecation warning like: "Please don't use this anymore but I cannot tell you what the future will bring"